### PR TITLE
e2e tests GCS additional test cases

### PR DESCRIPTION
**Pull request by sumitsri12**
@rmstar @itsankit-google Please review.

This PR depends on https://github.com/cdapio/cdap-e2e-tests/pull/75

cc @sawantpritam

### CODE REVIEW

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java [hunk 1]
```diff
@@ -215,4 +223,53 @@ public static void deleteTargetPubSubTopic() {
       }
     }
   }
+
+  /**
+   * Create GCS bucket with csv file containing below data types:
+   * int, float, string, double, long, timestamp, date, boolean, null, bytes.
```
Comment (rmstar) [inline src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java]: Where are we reading data as bytes, timestamps? Where are we reading null data; are there any nullable fields in your input?

File: src/e2e-test/java/io/cdap/plugin/gcs/stepsdesign/GCSBase.java [hunk 2]
```diff
@@ -194,4 +194,105 @@ public void verifyDataIsTransferredToTargetGCSBucket() {
...
+    if (verifyFilePresentInGcsBucket(gcsBucket, pathSuffix, "suffix")) {
+      BeforeActions.scenario.write("Data transferred to gcs bucket " + gcsBucket + " with pathSuffix "
+                                     + pathSuffix + " successfully");
+    } else {
+      Assert.fail("Data not transferred to target gcs bucket " + gcsBucket +
+                    " with pathSuffix " + pathSuffix);
+    }
+  }
+
+  private boolean verifyFilePresentInGcsBucket(String gcsBucket, String filename, String filePart) {
```
Comment (rmstar) [inline src/e2e-test/java/io/cdap/plugin/gcs/stepsdesign/GCSBase.java]: make filePart an enum so only specific values are allowed

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java (hunk 1 above)
Comment (sumitsri12) [reply src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java]: @rmstar We are reading these data from single csv file, which is in testdata folder, in the file we have Null field in which we are passing null values
src/e2e-test/resources/testdata/GCS_DATATYPE_TEST_1.csv

File: src/e2e-test/java/io/cdap/plugin/gcs/stepsdesign/GCSBase.java (hunk 2 above)
Comment (sumitsri12) [reply src/e2e-test/java/io/cdap/plugin/gcs/stepsdesign/GCSBase.java]: @rmstar filePart Enum created

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java (hunk 1 above)
Comment (rmstar) [reply src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java]: Looks like this is the schema for this file
```
gcsDataTypeTest1FileSchema=[{"key":"id","value":"int"},{"key":"name","value":"string"},\
  {"key":"yearofbirth","value":"int"},{"key":"isdeleted","value":"boolean"},{"key":"email","value":"string"},\
  {"key":"createddate","value":"string"},{"key":"revenue","value":"string"},{"key":"points","value":"string"},\
  {"key":"longdatatype","value":"string"},{"key":"doubledatatype","value":"double"},\
  {"key":"date","value":"string"},{"key":"null","value":"string"},{"key":"BytesData","value":"string"}]
```
Where are you testing bytes, timestamp types?

File: src/e2e-test/java/io/cdap/plugin/gcs/stepsdesign/GCSBase.java (hunk 2 above)
Comment (rmstar) [reply src/e2e-test/java/io/cdap/plugin/gcs/stepsdesign/GCSBase.java]: Type of filePart param should be FilePart, not string

Comment (ksahilCF) [reply src/e2e-test/java/io/cdap/plugin/gcs/stepsdesign/GCSBase.java]: Changed String param to FilePart

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java (hunk 1 above)
Comment (ksahilCF) [reply src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java]: Bug is already there. CSV is having data as timestamp. However, automatically it is taking as String. If we are overriding the data type then the pipeline is getting failed. Please refer bug- https://cdap.atlassian.net/browse/PLUGIN-1113

Comment (rmstar) [reply src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java]: I commented on the bug. Seems like input data in incorrect format (trying to read "2.01/0.0" as a float, "2021-09-20 11:27:50 UTC" as a long) ?

**Comment by ksahilCF** [comment]
Conflicts Resolved

**Comment by rmstar** [comment]
Please take a look at test failures

**Comment by sawantpritam** [comment]
> Please take a look at test failures

Conflicts resolved, Squashed commits, All tests passed.

**Review by rmstar — APPROVED** [review]

**Pull request merged**